server/src/test/java/org/apache/druid/server/http/DataSourcesResourceTest.java

```diff
@@ -830,7 +820,8 @@ public void testMarkAsUsedNonOvershadowedSegmentsNoDataSource()
         "datasource1",
         new DataSourcesResource.MarkDataSourceSegmentsPayload(Intervals.of("2010-01-22/P1D"), null)
     );
-    Assert.assertEquals(204, response.getStatus());
+    Assert.assertEquals(200, response.getStatus());
+    Assert.assertEquals(ImmutableMap.of("numChangedSegments", 0), response.getEntity());
     EasyMock.verify(segmentsMetadataManager);
   }
 
```

Comment:
Is there a similar test for `markAsUnused` API where there are no used segments or only used segments in the deep storage, and the operation still proceeds now that we have removed the short circuit? If not, I can add a test in this patch https://github.com/apache/druid/pull/16136

Reply:
added.